### PR TITLE
:sparkles: Add server status component

### PR DESCRIPTION
**Pull request by ibolton336**
- Disable analysis and getSolution if server is not running
- Move server status and prevent blocking analysis sidebar when server is starting
- Fix webview ready message to follow "type" convention
- Change startAnalysis to runAnalysis across all files

https://github.com/user-attachments/assets/d68bb713-8fde-482a-94fa-f4cbc3860086

### CODE REVIEW

File: vscode/src/client/analyzerClient.ts [hunk 1]
```diff
@@ -58,13 +58,14 @@ export class AnalyzerClient {
...
   public async start(): Promise<void> {
+    console.log("AnalyzerClient.start()");
     if (!this.canAnalyze()) {
       vscode.window.showErrorMessage("Cannot start the server due to missing configuration.");
       return;
     }
 
     try {
-      Promise.all([exec("java -version"), exec("mvn -version")]);
+      await Promise.all([exec("java -version"), exec("mvn -version")]);
```
Comment (rszwajko) [inline vscode/src/client/analyzerClient.ts:68]: good catch!

File: vscode/src/KonveyorGUIWebviewViewProvider.ts [hunk 2]
```diff
@@ -213,7 +213,7 @@ export class KonveyorGUIWebviewViewProvider implements WebviewViewProvider {
 
     webview.onDidReceiveMessage(
       (message) => {
-        if (message.command === "webviewReady") {
+        if (message.command === "WEBVIEW_READY") {
```
Comment (rszwajko) [inline vscode/src/KonveyorGUIWebviewViewProvider.ts:216]: 1. `message .type` ? 
2. webview seemed to work without this notification - do we still need it?

File: vscode/src/client/analyzerClient.ts [hunk 3]
```diff
@@ -276,11 +281,11 @@ export class AnalyzerClient {
         },
       );
 
-      console.log("response", response, incident, state);
-      vscode.commands.executeCommand("konveyor.loadSolution", response, { incident, violation });
+      vscode.commands.executeCommand("konveyor.loadSolution", response, {
```
Comment (rszwajko) [inline vscode/src/client/analyzerClient.ts:284]: it seems we have different formatting settings in the IDE

**Review by rszwajko — COMMENTED** [review]
Looks good !

File: vscode/src/KonveyorGUIWebviewViewProvider.ts (hunk 2 above)
Comment (ibolton336) [reply vscode/src/KonveyorGUIWebviewViewProvider.ts]: The webview_ready is used by the panels - need to look at this when fixing the fullscreen analysis page.

File: vscode/src/client/analyzerClient.ts (hunk 3 above)
Comment (ibolton336) [reply vscode/src/client/analyzerClient.ts:276]: Using prettier on my end. Not sure if there is a default prettier config that comes with the prettier plugin that may differ.

Comment (rszwajko) [reply vscode/src/client/analyzerClient.ts:273]: the line length is configured in the `.editorconfig` 
https://github.com/konveyor/editor-extensions/blob/9e3c6f3196a9fbb703124c57d403cd5ff15bd5bf/.editorconfig#L13

**Comment by rszwajko** [comment]
@ibolton336  
I wonder if we really need to surface the server state - we could keep it internal and start it automatically if user requests analysis to be run.
Looking at the wireframes we might even drop the button (I understand it would then be run automatically )

**Comment by sjd78** [comment]
I like having explicit status for the server available.  Back on the wireframes, I thought of it more as a status bar item with [wireframe 10](https://github.com/konveyor/editor-extensions/blob/main/docs/roadmap/wireframes/README.md#wf-10) (described but not drawn -- it is the blue bar in the status area in the wireframe image below).

[Wireframe 14](https://github.com/konveyor/editor-extensions/blob/main/docs/roadmap/wireframes/README.md#wf-14) has the progression from not configured forward...
image

In terms of UX, the server should manage itself like any other language server would.  If things are configured when the extension initializes, just start the server up right away.  The user changes the configs? Restart the server.  The extensions gets reloaded? Stop and start the server again.

For our purposes, manually starting and stopping the server can make a lot of sense as the start and initalize code can be changing.  Keeping those actions around and having a full set of server actions start, restart, and stop that can be hit from the command pallet is very vscode UX consistent.  Showing server status in the status bar is also far more vscode UX consistent.

**Review by djzager — APPROVED** [review]
ACK. I think we can iterate on this server status as we getting working on more systems more reliably in the coming weeks.

**Pull request merged**